### PR TITLE
Move CreateRelease into a Dialog component

### PR DESCRIPTION
**Pull request by hexagoncircle**
* Introduces _Dialog_ component that utilizes [Radix UI primitives](https://www.radix-ui.com/docs/primitives/components/dialog)
* Adds [slugify](https://www.npmjs.com/package/slugify) package
* Small refactor of _Account_, _Releases_, and a touch up to the _ReleaseCard_

### CODE REVIEW

File: components/Releases/CreateRelease.jsx [hunk 1]
```diff
@@ -1,5 +1,12 @@
+import slugify from "slugify"
 import { useState } from "react"
-import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { useUser, useSupabaseClient } from "@supabase/auth-helpers-react"
```
Comment (hexagoncircle) [inline components/Releases/CreateRelease.jsx:3]: We can grab the `user` from the `useUser` helper so you don't have to pass down the object into components.

File: components/Releases/CreateRelease.jsx [hunk 2]
```diff
@@ -10,46 +17,40 @@ const releaseTypes = [
...
-    user.user_metadata.type === "Label" ? user.user_metadata.name : null
+    user.user_metadata.type === "Label" ? user.user_metadata.name : ""
   )
   const [artworkUrl, setArtworkUrl] = useState()
   const [downloadUrl, setDownloadUrl] = useState()
   const [pagePassword, setPagePassword] = useState()
   const [isPasswordProtected, setIsPasswordProtected] = useState(false)
   const [type, setType] = useState(releaseTypes[5].text)
 
-  const [userId, setUserId] = useState(user.id)
```
Comment (hexagoncircle) [inline components/Releases/CreateRelease.jsx:28]: This isn't necessary, the `user.id` will never change

File: components/Releases/CreateRelease.jsx [hunk 3]
```diff
@@ -10,46 +17,40 @@ const releaseTypes = [
...
     title,
     artist,
     label,
     artworkUrl,
     downloadUrl,
     type,
-    userId,
   }) {
     try {
-      let slugger = title.toLowerCase()
```
Comment (hexagoncircle) [inline components/Releases/CreateRelease.jsx:40]: Replaced with `slugify` function

File: components/Releases/CreateRelease.jsx [hunk 4]
```diff
@@ -59,118 +60,117 @@ export default function CreateRelease({ user, setShowCreateNewRelease }) {
     } catch (error) {
       alert("Error creating new release!")
     } finally {
-      setShowCreateNewRelease(false)
+      console.log("All done!")
```
Comment (hexagoncircle) [inline components/Releases/CreateRelease.jsx:63]: This is where we will eventually programmatically close the dialog

**Review by frankmarra — APPROVED** [review]
💾

**Pull request merged**